apps/site/src/lib/components/forms/shared/ApplicationFlow.tsx

```diff
@@ -0,0 +1,60 @@
+import { PropsWithChildren } from "react";
+import { redirect } from "next/navigation";
+
+import hasDeadlinePassed from "@/lib/utils/hasDeadlinePassed";
+import haveApplicationsOpened from "@/lib/utils/haveApplicationsOpened";
+
+import ApplicationsClosed from "./ApplicationsClosed/ApplicationsClosed";
+import ApplyConfirm from "./ApplyConfirmation/ApplyConfirm";
+import Title from "./Title/Title";
+import getUserIdentity from "@/lib/utils/getUserIdentity";
+
+export const revalidate = 60;
+
+interface ApplicationFlowProps {
+	searchParams?: {
+		prefaceAccepted?: string;
+	};
```

Comment:
Thought: do you think the flow will need to know about other search params besides `prefaceAccepted`? There is some work going around in #446 on restoring URL path after logging in, so maybe keep for now and take a mental note to simplify later if not needed.

Reply:
Tracking in #480.